### ISSUE TITLE
amCharts is completely broken

### ISSUE THREAD

**Issue by Pauan** [codesandbox/codesandbox-client#4181]
# 🐛 bug report

## Description of the problem

When using amCharts it gives the following error:

```
TypeError
can't access property "requestFrame", system is undefined
evaluate/Sprite</Sprite.prototype.invalidate@https://k0n5k.csb.app/node_modules/
amcharts/amcharts4/.internal/core/Sprite.js:245:7
evaluate/Container</Container.prototype.invalidate@https://k0n5k.csb.app/node_modules/
amcharts/amcharts4/.internal/core/Container.js:127:33
Sprite@https://k0n5k.csb.app/node_modules/
amcharts/amcharts4/.internal/core/Sprite.js:184:11
Container@https://k0n5k.csb.app/node_modules/
amcharts/amcharts4/.internal/core/Container.js:48:24
createChild@https://k0n5k.csb.app/node_modules/
amcharts/amcharts4/.internal/core/utils/Instance.js:48:23
createFromConfig@https://k0n5k.csb.app/node_modules/
amcharts/amcharts4/.internal/core/utils/Instance.js:314:26
evaluate
/src/index.js:6:8

  3 | 
  4 | console.log(am4core.system.requestFrame);
  5 | 
> 6 | am4core.createFromConfig({}, "chartdiv", am4charts.XYChart);
    |        ^
  7 | 
  8 | console.log("Chart created");
  9 | 
```

But this shouldn't be possible, since [the erroring code](https://github.com/amcharts/amcharts4/blob/1d4fb42a5ecd71ce51dbf39e8852a57a3a447e3a/src/.internal/core/Sprite.ts#L1165) imports [system](https://github.com/amcharts/amcharts4/blob/1d4fb42a5ecd71ce51dbf39e8852a57a3a447e3a/src/.internal/core/Sprite.ts#L87), which is [*always* defined](https://github.com/amcharts/amcharts4/blob/1d4fb42a5ecd71ce51dbf39e8852a57a3a447e3a/src/.internal/core/System.ts#L537).

This error doesn't happen in regular webpages, and it also [doesn't happen with StackBlitz](https://stackblitz.com/edit/js-twtdaa).

So CodeSandbox is probably mangling the JS when transpiling.

### Your Environment

| Software         | Name/Version |
| ---------------- | ------------ |
| Сodesandbox      | 70a09ae7a
| Browser          | Firefox Dev Edition 77.0b8 (64-bit)
| Operating System | Windows 10 64-bit

**Comment by CompuIves** [comment]
Ah, I think I found the issue. Our es-module transpiler didn't handle a cyclic dependency well here, this will fix it: #4142. I'm going to merge part of that in today. Sorry for the inconvenience!

**Comment by Pauan** [comment]
@CompuIves Great, I really appreciate the fast response!

**Comment by CompuIves** [comment]
The fix for this should be live within the next 20 minutes!

**Comment by pete-redmond-cko** [comment]
@CompuIves and @Pauan I think amcharts (at least v4) is still completely broken in codesandbox


I am unable to use amcharts in codesandbox (https://codesandbox.io/s/competent-smoke-2mz6q) , I always get the following error

```
TypeError
Object prototype may only be an Object or null: undefined
```

Searching through codesandbox I can this issue is widespread (the links below are from the first 2 pages of search results)

https://codesandbox.io/s/pedantic-blackburn-eowux
https://codesandbox.io/s/competent-leavitt-v11vz
https://codesandbox.io/s/musing-khayyam-8i4pr
https://codesandbox.io/s/lively-dew-mpb25
https://codesandbox.io/s/currying-framework-bz8mh
https://codesandbox.io/s/lucid-bas-mt652
https://codesandbox.io/s/reactchart-b17co

all with the same issue

**Comment by ACVis** [comment]
I can confirm amcharts4 is still completely broken in Codesandbox.

**Comment by lbogdan** [comment]
Hey @ACVis ,

Can you please provide a sandbox URL where it's broken? For me the previous comment "broken" sandbox (https://codesandbox.io/s/fancy-frog-bed8w) works fine.

**Comment by ACVis** [comment]
I swear it wasn't working for the last two weeks, I had dedicated pens just for showing the error and spent hours trying to fix that exact error message, but it seems to be working now. Coincidentally I had talked to the Amcharts guys too and they said this had happened before and must be a regression and that they would message someone at Codesandbox? Either way, it is working now and definitely was not before.
